### ISSUE TITLE
Black Pixels in Converted GIF

### ISSUE THREAD

**Issue by ayush-sharma11** [caed0/webp-conv#2]
The conversion process from WEBP to GIF completes successfully. However, the resulting GIF contains numerous random black pixels, impacting the output's quality.
Example of a converted gif image having the issue:
![output](https://github.com/user-attachments/assets/817cc637-5bf7-45ef-8d32-8de6658a479c)

I tried changing the optional quality value from 0 to 10 to even 100 but the issue is still there.

Thank you

**Comment by caed0** [comment]
can you give me the original webp file so i can try to replicate it?

**Comment by ayush-sharma11** [comment]
yes. this is the one i used: https://giphy.com/gifs/theoffice-the-office-tv-lecture-circuit-part-1-66dLOWrLqrgxWHSeVR

**Comment by caed0** [comment]
the issue seems to be the compression of some webp files i am working on a solution

**Comment by ayush-sharma11** [comment]
thank you. i really appreciate your help.

**Comment by caed0** [comment]
it's probably too late, but i just haven't had time recently. it should be working properly now. the issue was due to the way frame extraction worked, as some webp files store only the differences from the previous frame, which produced these artifacts. the code now uses libwebp to accurately extract and decode frames.

**Comment by ayush-sharma11** [comment]
thank you, mb i took forever to reply. i'll check it now and let you know if there's any issue.
